ros2component/test/test_cli_load.py

```diff
@@ -0,0 +1,196 @@
+                    cmd=['ros2', 'daemon', 'start'],
+                    name='daemon-start',
+                    on_exit=[
+                        component_node,
+                        OpaqueFunction(function=lambda context: ready_fn())
+                    ],
+                    additional_env=additional_env
+                )
+            ]
+        ),
+    ]), locals()
+
+
+class TestROS2ComponentLoadCLI(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(
+        cls,
+        launch_service,
+        proc_info,
+        proc_output,
+        rmw_implementation
+    ):
+        @contextlib.contextmanager
+        def launch_node_command(self, arguments):
+            node_command_action = ExecuteProcess(
+                cmd=['ros2', 'component', *arguments],
+                additional_env={
+                    'RMW_IMPLEMENTATION': rmw_implementation,
+                    'PYTHONUNBUFFERED': '1'
+                },
+                name='ros2component-cli',
+                output='screen'
+            )
+            with launch_testing.tools.launch_process(
+                launch_service, node_command_action, proc_info, proc_output,
+                output_filter=launch_testing_ros.tools.basic_output_filter(
+                    # ignore ros2cli daemon nodes
+                    filtered_patterns=['.*ros2cli.*'],
+                    filtered_rmw_implementation=rmw_implementation
+                )
+            ) as node_command:
+                yield node_command
+        cls.launch_node_command = launch_node_command
+
+    @launch_testing.markers.retry_on_failure(times=2)
+    def test_load_verb(self):
+        with self.launch_node_command(
+                arguments=[
+                    'load', '/ComponentManager',
+                    'ros2component_fixtures', 'ros2component_fixtures::Talker']) as talker_node:
+            assert talker_node.wait_for_shutdown(timeout=20)
+        assert talker_node.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=[
+                "Loaded component 1 into '/ComponentManager' "
+                "container node as '/talker'"],
+            text=talker_node.output,
+            strict=True
+        )
+        with self.launch_node_command(
+                arguments=[
+                    'load', '/ComponentManager',
+                    'ros2component_fixtures',
+                    'ros2component_fixtures::Listener']) as listener_node:
+            assert listener_node.wait_for_shutdown(timeout=20)
+        assert listener_node.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=[
+                "Loaded component 2 into '/ComponentManager' "
+                "container node as '/listener'"],
+            text=listener_node.output,
+            strict=True
+        )
+        with self.launch_node_command(
```

Comment:
@BMarchi I think it'd be good to assert on container output to make sure components were indeed loaded.

Reply:
That implies that I'll have to deal with discovery delay and related stuff

Reply:
And to check that I'll need to get the output from the container that I'm loading in `generate_test_description` process.

Reply:
> This should only test the cli, not if the node is working as it should, that should be tested on its own place.

I fully agree, but that applies to both talker and/or listener components (and thus my comment, https://github.com/ros2/ros2cli/pull/390#discussion_r343117111). Checking the container output, assuming we can trust it (and that's being tested elsewhere), is the simplest way to make sure that `load` and `unload` actually worked. Otherwise you're not checking if the CLI has the expected side effect.